### ISSUE TITLE
Suggest warning for unnecessary braces

### ISSUE THREAD

**Issue by bje-** [koalaman/shellcheck#1422]
#### For new checks and feature suggestions
- [X ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [X ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/your/interpreter
foo=1
echo ${foo}
```

#### Here's what shellcheck currently says:

Nothing.

#### Here's what I wanted or expected to see:

Unnecessary braces around $foo. Many users place unnecessary variable expansions around variable names that do not require them. For example, if the variable number contains an underscore.

**Comment by orbea** [comment]
What is the harm in adding such braces? This is a stylistic choice and shellcheck should not dictate style.

**Comment by bje-** [comment]
Agreed, but the extra syntactic sugar clutters the code and makes it harder to read (in most circumstances). eg `${foobar}` versus just `$foobar`. The harm is that novice shell programmers just copy what they see in existing shell code without ever understanding the quoting rules. It would be good if interested users could ask Shellcheck to warn about such style issues.

**Comment by orbea** [comment]
Shellcheck already warns a lot about incorrect quoting and the consequence of this proposal would be countless incorrect warnings for correct code, this would not encourage novice users for writing better shell more than it would encourage more experienced users from not using shellcheck.

Additionally based upon the font, editor, personal preference and other stylistic choices made in the script you have no way of actually knowing what would be easier to read for any single person.
